models/models.go

```diff
@@ -24,3 +28,224 @@ func standardizeError(err error) error {
 
 	return err
 }
+
+//This is an example of a structured format that is compatible with Object-Relational Mapping (ORM) systems.
+type MyStruct struct {
+	Id          int64  `json:"id" db:"id" primarykey:"true" `
+	Name        string `json:"name" db:"name" `
+	Age         int    `json:"age" db:"age"`
+	Email       string `json:"email" db:"email"`
+	PhoneNumber int64  `json:"phone_number" db:"phone_number"`
+}
+
+//structure variable is having the &struct.please pass the pointer of the structure not the variable
+func FindFirst(tableName string, structure interface{}) error {
+	primaryKeyField, err := PrimaryKeyIdentifier(structure)
+	if err != nil {
+		return err
+	}
+	// Build the query using the determined primary key field.
+	query := fmt.Sprintf("SELECT * FROM %s ORDER BY %s LIMIT 1", tableName, primaryKeyField)
+	// Execute the query and scan the result into the provided structure.
+	errFromDb := utility.Db.Get(structure, query)
+	return errFromDb
+}
+
+//structure variable is having the &struct.please pass the pointer of the structure not the variable
+func FindLast(tableName string, structure interface{}) error {
+	primaryKeyField, err := PrimaryKeyIdentifier(structure)
+	if err != nil {
+		return err
+	}
+	query := fmt.Sprintf("SELECT * FROM %s ORDER BY  %s DESC LIMIT 1", tableName, primaryKeyField)
+	errFromDb := utility.Db.Get(structure, query)
+	return errFromDb
+}
+
+//data varibale need to have the three compulsory keys (tablename,columnname,columnvalue),structure variable is having the &[]struct
+func Find(data map[string]interface{}, structure interface{}) error {
+	// Check if the required keys exist in the 'data' map.
+	tableName, tableNameExists := data["tablename"].(string)
+	columnName, columnNameExists := data["columnname"].(string)
+	columnValue, columnValueExists := data["columnvalue"]
+	sortColumn, sortColumnExists := data["sortcolumn"].(string)
+	sortValue, sortValueExists := data["sortvalue"].(string)
+
+	// Check if the required keys are missing and handle the error condition.
+	if !tableNameExists || !columnNameExists || !columnValueExists {
+		return errors.New("missing required keys in map[string]interface{} any of this tablename,columnname,columnvalue")
```

Comment:
check with the team, I think the errors can be a bit more simpler and easier to understand